pytest_girder/pytest_girder/fixtures.py

```diff
@@ -129,3 +155,47 @@ def user(db, admin):
     yield u
 
     User().remove(u)
+
+
+@pytest.fixture
+def fsAssetstore(db, request):
+    """
+    Require a filesystem assetstore. Its location will be derived from the test function name.
+    """
+    from girder.constants import ROOT_DIR
+    from girder.models.assetstore import Assetstore
+
+    name = '_'.join((
+        request.node.module.__name__,
+        request.node.cls.__name__ if request.node.cls else '',
+        request.node.name))
+
+    path = os.path.join(ROOT_DIR, 'tests', 'assetstore', name)
+
+    if os.path.isdir(path):
+        shutil.rmtree(path)
+
+    yield Assetstore().createFilesystemAssetstore(name=name, root=path)
+
+    if os.path.isdir(path):
+        shutil.rmtree(path)
+
+
+@pytest.fixture
+def testPlugins():
+    """
+    Make plugins load from the test_plugins dir rather than the normal location.
+    """
+    from girder.constants import ROOT_DIR
+    from girder.utility import plugin_utilities
+
+    path = os.path.join(ROOT_DIR, 'tests', 'test_plugins')
+    old = plugin_utilities.getPluginDir
+    plugin_utilities.getPluginDir = mock.Mock(return_value=path)
+
+    yield path
+
+    plugin_utilities.getPluginDir = old
```

Comment:
We haven't solved the problem of testing plugins yet.  I hesitate to bake in the assumption that plugin tests exist inside girder's package.  For girder 3, plugin tests will be include inside each plugin's package.  What we do before then is up in the air.  Because this isn't used yet, could you remove it from this PR?

Reply:
This isn't for testing plugins, it's for loading the fake plugins that we use in core testing. Its only contractual guarantee is that it will cause plugins to be loaded from the fake directory instead of the real one.

Reply:
(Also it's used currently in the mail test)

Reply:
Okay, I see.  We'll probably need to revisit this for the pip installable plugins changes, but for now I think it's okay.

Reply:
Yeah, since it's just for testing core and not plugins, I don't care if we need to break it later since it'll just be us having to deal with that.